vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -450,16 +443,111 @@ public void setItems(ItemFilter<T> itemFilter, Collection<T> items) {
      * the current item set without resetting the value, you should update the
      * previously set item collection and call
      * {@code getDataProvider().refreshAll()}.
+     * <p>
+     * The returned data view object can be used for further access to ComboBox
+     * items, or later on fetched with {@link #getListDataView()}. For using
+     * lazy data load, use one of the {@code setItemsWithFilter} instead.
      *
      * @param itemFilter
      *            filter to check if an item is shown when user typed some text
      *            into the ComboBox
      * @param items
      *            the data items to display
+     *
+     * @return the in-memory data view instance that provides access to the data
+     *         bound to the ComboBox
      */
-    public void setItems(ItemFilter<T> itemFilter,
+    public ComboBoxListDataView<T> setItems(ItemFilter<T> itemFilter,
             @SuppressWarnings("unchecked") T... items) {
-        setItems(itemFilter, Arrays.asList(items));
+        return setItems(itemFilter, Arrays.asList(items));
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @deprecated Because the stream is collected to a list anyway, use
+     *             {@link HasListDataView#setItems(Collection)} or
+     *             {@link #setItems(CallbackDataProvider.FetchCallback)}
+     *             instead.
+     */
+    @Deprecated
+    public void setItems(Stream<T> streamOfItems) {
+        setItems(DataProvider.fromStream(streamOfItems));
+    }
+
+
+    @Override
+    public ComboBoxDataView<T> setItems(DataProvider<T, String> dataProvider) {
+        setDataProvider(dataProvider);
+        return getGenericDataView();
+    }
+
+    /**
+     * Gets the generic data view for the ComboBox. This data view should
+     * only be used when {@link #getListDataView()} or
+     * {@link #getLazyDataView()} are not applicable for the underlying data
+     * provider.
+     *
+     * @return the generic {@link DataView} implementation for ComboBox
+     * @see #getListDataView()
+     * @see #getLazyDataView()
+     */
+    @Override
+    public ComboBoxDataView<T> getGenericDataView() {
+        return new ComboBoxDataView<>(getDataCommunicator(), this);
+    }
+
+    @Override
+    public ComboBoxLazyDataView<T> setItems(BackEndDataProvider<T, String> dataProvider) {
+        setDataProvider(dataProvider);
+        return getLazyDataView();
+    }
+
+    /**
+     * Gets the lazy data view for the ComboBox. This data view should only be
+     * used when the items are provided lazily from the backend with:
+     * <ul>
+     * <li>{@link #setItems(CallbackDataProvider.FetchCallback)}</li>
+     * <li>{@link #setItems(CallbackDataProvider.FetchCallback, CallbackDataProvider.CountCallback)}</li>
+     * <li>{@link #setItems(BackEndDataProvider)}</li>
```

Comment:
Missing the API which accepts callback + filter converter ?

Reply:
Done